lib/external-housing-available/external-housing-available.controllers.js

```diff
@@ -1,6 +1,172 @@
+            severity: 'Notice',
+            request: request,
+          });
+        } catch (e) {}
+        throw Boom.badData('No results for that address');
+      } 
+
+      const firstGeocode = geocodes.results[0];
+
+      let addressObject = postalAddressLib.extractAddress(firstGeocode);
+
+      if (!addressObject.addressNumber) {
+        try {
+          await Log.request({
+            type: 'externalhousingaddress.submit.failed.street_number',
+            severity: 'Notice',
+            request: request,
+          });
+        } catch (e) {}
+        throw Boom.badData('Address is missing a street number');
+      }
+
+      //Address without a county, we want to use locality instead.
+      if (!addressObject.hasOwnProperty('addressCounty')) {
+        addressObject.addressCounty = addressObject.addressLocality;
+      }
+      if (
+        addressObject.addressRegion !== process.env.ADDRESS_LIMIT_STATE ||
+        addressObject.addressCounty !== process.env.ADDRESS_LIMIT_COUNTY
+      ) {
+        try {
+          await Log.request({
+            type: 'externalhousingaddress.submit.failed.county',
+            severity: 'Notice',
+            request: request,
+          }); 
+        } catch (e) {} 
+        throw Boom.badData('Sorry, We are only accepting properties in ' + `${process.env.ADDRESS_LIMIT_COUNTY}, ${process.env.ADDRESS_LIMIT_STATE}`); 
+      }
+
+      //Extracting the address using function from geoCoordinates lib
+      let geoCoordObject = geoCoordinatesLib.extractLocation(firstGeocode);
+      
+      const addressHash = postalAddressLib.hashAddress(addressObject);
+      addressObject.hash = addressHash;
+
+      // External Housing Available Object 
+      let externalHousingAvailableObject = {};
+
+      const existingAddresses = await postalAddressLib.getAddressesByHash(addressHash);
+      if (existingAddresses.length > 0) {
+        //If this is an exisiting address
+        externalHousingAvailableObject.PropertyId = existingAddresses[0].PropertyId;
+      } else {
+        //Else, create a new propertyId for non exisiting address
+      }
+
+      const standardizedAddress = postalAddressLib.standardizeAddress(addressObject);
+      addressObject.address = standardizedAddress; 
+
+      //Title
+      if (request.payload.title && request.payload.title !== request.payload.address) {
+        externalHousingAvailableObject.title = request.payload.title;
+      } else {
+        externalHousingAvailableObject.title = standardizedAddress; 
+      }
+      //AuthorId
+      if (request.payload.AuthorId && request.auth.credentials.subjectId === process.env.SUPER_ADMIN) {
+        externalHousingAvailableObject.AuthorId = request.payload.AuthorId;
+      } else {
+        throw Boom.badRequest('ExternalHousingAvailable.AuthorId must match an Super Admin account ID.');
+      }
+      //Address
+      if (standardizedAddress) {
+        externalHousingAvailableObject.address = standardizedAddress;
```

Comment:
let's keep the address as address, instead of standardized address. as mentioned, standardized has unit info stripped out, which we don't want done in this case.